From 1d15e3b288d222a021a3369b2ec1ce4549a7c049 Mon Sep 17 00:00:00 2001
From: Alexei Colin <alexei.co@gmail.com>
Date: Sun, 16 Dec 2012 22:19:58 -0500
Subject: [PATCH] Connector 2.1: expose attachmentMode in saveItems

This allows one to add files by path via the connector.
---
 chrome/content/zotero/xpcom/server_connector.js |   20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/chrome/content/zotero/xpcom/server_connector.js b/chrome/content/zotero/xpcom/server_connector.js
index e8312cb..065a512 100755
--- a/chrome/content/zotero/xpcom/server_connector.js
+++ b/chrome/content/zotero/xpcom/server_connector.js
@@ -22,7 +22,7 @@
     
     ***** END LICENSE BLOCK *****
 */
-const CONNECTOR_API_VERSION = 2;
+const CONNECTOR_API_VERSION = 2.1;
 
 Zotero.Server.Connector = function() {};
 Zotero.Server.Connector._waitingForSelection = {};
@@ -268,6 +268,9 @@ Zotero.Server.Connector.SavePage.prototype = {
  *
  * Accepts:
  *		items - an array of JSON format items
+ *		attachmentMode - add items as files ('file') or download ('download'),
+ *					     optional, default is 'download'
+ *					     (see ItemSaver.ATTACHMENT_MODE_*)
  * Returns:
  *		201 response code with empty body
  */
@@ -295,10 +298,21 @@ Zotero.Server.Connector.SaveItem.prototype = {
 		
 		var cookieSandbox = data["uri"] ? new Zotero.CookieSandbox(null, data["uri"],
 			data["cookie"] || "", url.userAgent) : null;
-		
+
+		var attachmentMode =
+			Zotero.Translate.ItemSaver.ATTACHMENT_MODE_DOWNLOAD;
+		var requestedAttachmentMode = data["attachmentMode"];
+		if (requestedAttachmentMode && requestedAttachmentMode.toUpperCase) {
+			var attachmentModeName =
+				"ATTACHMENT_MODE_" + requestedAttachmentMode.toUpperCase();
+			if (Zotero.Translate.ItemSaver[attachmentModeName]) {
+				attachmentMode = Zotero.Translate.ItemSaver[attachmentModeName];
+			}
+		}
+
 		// save items
 		var itemSaver = new Zotero.Translate.ItemSaver(libraryID,
-			Zotero.Translate.ItemSaver.ATTACHMENT_MODE_DOWNLOAD, 1, undefined, cookieSandbox);
+			attachmentMode, 1, undefined, cookieSandbox);
 		itemSaver.saveItems(data.items, function(returnValue, data) {
 			if(returnValue) {
 				try {
-- 
1.7.9.5

